ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PreemptableSequence.scala

```diff
@@ -54,13 +54,13 @@ trait SequenceHelper {
 
   /** Wrap a synchronous call into a Future sequence, which
     * - will be preemptable
-    * - will retry to execute the body if Exception-s thrown
+    * - will retry to execute the body if Exception-s thrown, and retry guard allows it
     *
     * @return the preemptable, retrying Future sequence
     */
   def retry[T](waitMillisBetweenRetries: Long, maxAmountOfRetries: Long = -1)(
       body: => T
-  ): Future[T]
+  )(retryGuard: Throwable => Boolean): Future[T]
```

Comment:
It is not clear from name if true means retry or don't retry.  Maybe rename to `retryableException`, or clarify in scaladoc.

Reply:
good idea, I like simply retryable